tests/testthat/test-parameters.R

```diff
@@ -0,0 +1,403 @@
+    )
+    write_sample_parameters_file(
+      value = expected,
+      parameter = "delay",
+      path = "delay_interval.parquet",
+      disease = disease,
+      state = NA,
+      start_date = start_date,
+      end_date = NA
+    )
+    write_sample_parameters_file(
+      value = expected,
+      parameter = "right_truncation",
+      path = "right_truncation.parquet",
+      disease = disease,
+      state = "test",
+      start_date = start_date,
+      end_date = NA
+    )
+
+
+    actual <- read_parameters(
+      generation_interval_path = "generation_interval.parquet",
+      delay_interval_path = NULL,
+      right_truncation_path = NULL,
+      disease = "COVID-19",
+      as_of_date = start_date + 1,
+      state = "test"
+    )
+  })
+
+
+  expect_equal(
+    actual,
+    list(
+      generation_interval = expected,
+      delay_interval = NA,
+      right_truncation = NA
+    )
+  )
+})
+
+test_that("Can read right-truncation on happy path", {
+  expected <- c(0.8, 0.2)
+  path <- "test.parquet"
+  parameter <- "right_truncation"
+  start_date <- as.Date("2023-01-01")
+
+  # COVID-19
+  disease <- "COVID-19"
+  withr::with_tempdir({
+    write_sample_parameters_file(
+      value = expected,
+      path = path,
+      state = "test",
+      disease = disease,
+      parameter = parameter,
+      start_date = start_date,
+      end_date = NA
+    )
+    actual <- read_interval_pmf(
+      path = path,
+      parameter = parameter,
+      disease = disease,
+      as_of_date = start_date + 1,
+      state = "test"
+    )
+  })
+  expect_equal(actual, expected)
+
+
+  # Influenza
+  disease <- "Influenza"
+  withr::with_tempdir({
+    write_sample_parameters_file(
+      value = expected,
+      path = path,
+      state = "test",
+      disease = disease,
+      parameter = parameter,
+      param = parameter,
+      start_date = start_date,
+      end_date = NA
+    )
+    actual <- read_interval_pmf(
+      path = path,
+      parameter = parameter,
+      disease = disease,
+      as_of_date = start_date + 1,
+      state = "test"
+    )
+  })
+  expect_equal(actual, expected)
+})
+
+test_that("Invalid PMF errors", {
+  expected <- c(0.8, -0.1)
+  path <- "test.parquet"
+  parameter <- "right_truncation"
+  start_date <- as.Date("2023-01-01")
+
+  # COVID-19
+  disease <- "COVID-19"
+  withr::with_tempdir({
+    write_sample_parameters_file(
+      value = expected,
+      path = path,
+      state = "test",
+      disease = disease,
+      parameter = parameter,
+      param = parameter,
+      start_date = start_date,
+      end_date = NA
+    )
+    expect_error(read_interval_pmf(
+      path = path,
+      parameter = parameter,
+      disease = disease,
+      as_of_date = start_date + 1,
+      state = "test"
+    ))
+  })
+})
+
+
+test_that("Can read delay on happy path", {
+  expected <- c(0.8, 0.2)
+  path <- "test.parquet"
+  parameter <- "delay"
+  start_date <- as.Date("2023-01-01")
+
+  # COVID-19
```

Comment:
Is this exactly the same as the influenza test aside from a string? Make a function?

Reply:
It is! But I don't think it should be a function. I don't think un-DRY tests are a bad thing. They make it easy to see clearly what's happening, what's changing, and what the state is under test. They also make it easy to debug because you can step through all the individual pieces, without relying on an external function.

I've extracted out some simple boilerplate into a helper to set up the test, but I don't think we want the actual test itself to be dropped into a function.